src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +586,32 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        // Get all valid options from the configuration files
+        var validArguments = app.GetOptions().SelectMany(o => new[] { o.LongName, o.ShortName }).ToArray();
+
+        var argumentsNamesProvided = GetArgumentNames(args).Select(a => a.ToString()).ToArray();
+
+        foreach (var argumentName in argumentsNamesProvided)
+        {
+            // Check if the argument provided is a valid option/argument
+            if (!validArguments.Contains(argumentName))
+            {
+                string message = $"Unrecognized argument: {argumentName}.\nRun --help for a list of available options and commands.";
+                throw new ArgumentException(message);
+            }
+        }
+
+        string duplicateArgumentsList = string.Join(", ", GetDuplicateArguments(args));
```

Comment:
Don't pass args here again, pass `argumentsNamesProvided`

Reply:
Good call
No need to process/get the names again if we already did it